### PR TITLE
Issue 106 Fixes

### PR DESCRIPTION
**Pull request by maxpohlman**
This pull addresses the issues discussed in issue #106.

There are currently inconsistencies in the naming of the data structures returned by `fetch()`. Datasets without hyphens in their name return the indexed data frames’ names without the original dataset name (`data<-fetch(‘portal’)` yields `data$main`) while datasets with hyphens in their name return data frames with the dataset name (`data<-fetch(‘mammal-community-db’)` yields `data$mammal_community_db_species` instead of `data$species`).

The proposed solution by @dmcglinn of deleting line 102 does not solve that issue. That line is important to the renaming structure as it removes the dataset name ‘header’ from non-hyphenated datasets (in the example above, without line 102, there would be `data$portal_main`). The reason it does not remove underscores from datasets like `mammal-community-db` is because it is attempting to search for the dataset name (which contains hyphens) in `list_names`, which has already converted the hyphens to underscores in line 98. Thus, it can’t find the hyphenated name within the underscored name and the underscores are not substituted.

A solution is changing line 102 to instead search using `dataset_underscores`. This now searches with the converted-to-underscore name in `list_names`. After the change, `fetch(‘portal’)` and `fetch(‘mammal-community-db’)` both yield lists that do not contain the dataset name.

In addition, I also implemented the suggestion by @ethanwhite to allow `fetch()` to accept multiple datasets and return them in a nested list. I do this by having the function create a `master` list and run a for loop for each dataset inputted, then store the output into a master list. Each dataset stored into the master list is stored under the original name of the dataset, but the master is what is returned and thus can be assigned to whatever name you want. Finally, in order to avoid unnecessary nesting, the function does not return the master dataset if the user only inputs one dataset and simply acts as it did before (although with the naming changes above).

In summary:
Old: `data<-fetch('portal')` yields `data$main`
New: `data<-fetch('portal')` yields `data$main`

Old: `data<-fetch(‘mammal-community-db’)` yields `data$mammal_community_db_species`
New: `data<-fetch(‘mammal-community-db’)` yields `data$species`

Old: `data<-fetch(c(‘mammal-community-db’,'portal))` yields `error`
New:  `data<-fetch(c(‘mammal-community-db’,'portal))` yields `data$portal$main` and `data$mammal_community_db$species`

### CODE REVIEW

File: R/rdataretriever.R [hunk 1]
```diff
@@ -88,22 +88,40 @@ install = function(dataset, connection, db_file=NULL, conn_file=NULL,
 #' }
 fetch = function(dataset, quiet=TRUE){
   temp_path = tempdir()
-  if (quiet)
+  master = vector("list", length(dataset))
+  names(master) = dataset
+  names(master) = gsub("-","_",names(master))
+  z = 1
```
Comment (henrykironde) [inline R/rdataretriever.R]: Thank you for the contribution @maxpohlman. It would be nice to have the variables name reflecting what they store for easy readability.

**Comment by maxpohlman** [comment]
The naming of the nested data frames takes on the default name of the dataset (but with the hyphens converted to underscores). I just added a commit that would allow users to specify the names of those data frames through a `dnames=` argument. The default would still be the default dataset names, but now the users have extra customization options and would help with longer dataset names.

Examples:
 `data<-fetch(c(‘mammal-community-db’,'portal))` yields `data$portal$main` and `data$mammal_community_db$species`

`data<-fetch(c(‘mammal-community-db’,'portal), dnames=c('nameone','nametwo'))` yields `data$nameone$main` and `data$nametwo$species`

**Comment by dmcglinn** [comment]
Thanks for this great contribution @maxpohlman. I will try to review it this weekend.

**Comment by dmcglinn** [comment]
I apologize for the huge delay on testing this. I tested the code it works great. @henrykironde thanks for the notes on refactoring the code. I'm going to merge @maxpohlman 's great contribution and then I'll do a little bit of refactoring work. Thanks again @maxpohlman would you like to be listed as a co-author on the package. If so please share your email address.

**Comment by maxpohlman** [comment]
@dmcglinn Happy to help. Sure, my email is maxpohlman@gmail.com
